metadata-ingestion/src/datahub/ingestion/api/workunit.py

```diff
@@ -97,10 +97,15 @@ def get_aspects_of_type(self, aspect_cls: Type[T_Aspect]) -> List[T_Aspect]:
         elif isinstance(self.metadata, MetadataChangeProposalWrapper):
             aspects = [self.metadata.aspect]
         elif isinstance(self.metadata, MetadataChangeProposal):
-            # Can't get aspects from MetadataChangeProposal
-            # It seems this is only used by mcp_patch_builder
-            # So this method does not collect patches
             aspects = []
+            # Best effort attempt to deserialize MetadataChangeProposalClass
+            if self.metadata.aspectName == aspect_cls.ASPECT_NAME:
+                try:
+                    mcp = MetadataChangeProposalWrapper.from_mcpc(self.metadata)
+                    if mcp:
+                        aspects = [mcp.aspect]
+                except Exception:
```

Comment:
this try catch is redundant

Reply:
Is it? There's no try catch in `from_mcpc` or `_try_from_generic_aspect`